### PR TITLE
Improving HP support

### PR DESCRIPTION
**Pull request by ErwanAliasr1**
This PR is about improving the detection of HP Servers.

### CODE REVIEW

**Comment by ErwanAliasr1** [comment]
@dtantsur any comments on this one ?

File: hardware/bios_hp.py [hunk 1]
```diff
@@ -27,38 +27,33 @@
...
-        output_file = next(tempfile._get_candidate_names())
-        status, output = cmd("hp-conrep --save -f {}".format(output_file))
-        if status != 0:
-            sys.stderr.write("Unable to run hp-conrep: %s\n" % output)
-            return False
-        xml = ET.parse(output_file)
-        os.remove(output_file)
-        root = xml.getroot().iter("Section")
+    valid, hp_config = get_hp_conrep(hw_lst)
+    if valid is False:
```
Comment (dtantsur) [inline hardware/bios_hp.py]: nit: `if not valid`

File: hardware/bios_hp.py [hunk 2]
```diff
@@ -27,38 +27,33 @@
...
+        return False
+
+    if len(hp_config) > 0:
```
Comment (dtantsur) [inline hardware/bios_hp.py]: nit: `if hp_config`

File: hardware/bios_hp.py (hunk 1 above)
Comment (ErwanAliasr1) [reply hardware/bios_hp.py]: fixed.

File: hardware/bios_hp.py (hunk 2 above)
Comment (ErwanAliasr1) [reply hardware/bios_hp.py]: fixedt.

**Pull request merged**